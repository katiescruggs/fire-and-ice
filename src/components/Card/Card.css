.card {
  border: 2px solid black;
  border-radius: 10px;
  margin: 10px;
  padding: 5px;
  width: 300px;
}

.card h3 {
  font-size: 34px;
  margin: 15px;
}

.card .words {
  font-size: 24px;
}

.card .founded {
  font-size: 20px;
}

.hide {
  display: none;
}