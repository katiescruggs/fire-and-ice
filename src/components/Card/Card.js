import React from 'react';
import './Card.css';
import { fetchMembers } from '../../helpers/apiCalls';

const getMemberArray = async (swornMembers) => {
  const members = await fetchMembers(swornMembers);
  console.log(members);
}


const Card = ({ house }) => {
  const {name, founded, seats, titles, coatOfArms, ancestralWeapons, words} = house;
  const displayFounded = founded ? founded : 'N/A';
  const swornMembers = getMemberArray(house.swornMembers);
  
  return (
    <div className="card">
      <h3>{name}</h3>
      <h4>{words}</h4>
      <h4>Founded: {displayFounded}</h4>
      <p>Seats: {seats}</p>
      <p>Titles: {titles}</p>
      <p>Ancestral Weapons: {ancestralWeapons}</p>
      <p>Coat of Arms: {coatOfArms}</p>
    </div>
  );
}

export default Card;