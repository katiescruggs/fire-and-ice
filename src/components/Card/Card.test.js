import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let card;

  window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
    json: () => Promise.resolve({
      member: 'I am a sworn member!'
    })
  }));
  const mockHouse = {
    name: 'NAME',
    founded: 'FOUNDED',
    seats: 'SEATS',
    titles: 'TITLES',
    coatOfArms: 'COATOFARMS',
    ancestralWeapons: 'ANCESTRALWEAPONS',
    words: 'WORDS',
    swornMembers: ['array', 'of', 'api', 'urls']
  };

  beforeEach(() => {
    card = shallow(<Card house={mockHouse} />);
  });

  it('should exist', () => {
    expect(card).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(card).toMatchSnapshot();
  });
});