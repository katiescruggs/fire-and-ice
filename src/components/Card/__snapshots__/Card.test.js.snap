// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": "ANCESTRALWEAPONS",
            "coatOfArms": "COATOFARMS",
            "founded": "FOUNDED",
            "name": "NAME",
            "seats": "SEATS",
            "swornMembers": Array [
              "array",
              "of",
              "api",
              "urls",
            ],
            "titles": "TITLES",
            "words": "WORDS",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          NAME
</h3>,
        <h4>
          WORDS
</h4>,
        <h4>
          Founded: 
          FOUNDED
</h4>,
        <p>
          Seats: 
          SEATS
</p>,
        <p>
          Titles: 
          TITLES
</p>,
        <p>
          Ancestral Weapons: 
          ANCESTRALWEAPONS
</p>,
        <p>
          Coat of Arms: 
          COATOFARMS
</p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NAME",
        },
        "ref": null,
        "rendered": "NAME",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "WORDS",
        },
        "ref": null,
        "rendered": "WORDS",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "FOUNDED",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "FOUNDED",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            "SEATS",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          "SEATS",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles: ",
            "TITLES",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Titles: ",
          "TITLES",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            "ANCESTRALWEAPONS",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          "ANCESTRALWEAPONS",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            "COATOFARMS",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          "COATOFARMS",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            NAME
</h3>,
          <h4>
            WORDS
</h4>,
          <h4>
            Founded: 
            FOUNDED
</h4>,
          <p>
            Seats: 
            SEATS
</p>,
          <p>
            Titles: 
            TITLES
</p>,
          <p>
            Ancestral Weapons: 
            ANCESTRALWEAPONS
</p>,
          <p>
            Coat of Arms: 
            COATOFARMS
</p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NAME",
          },
          "ref": null,
          "rendered": "NAME",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WORDS",
          },
          "ref": null,
          "rendered": "WORDS",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "FOUNDED",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "FOUNDED",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              "SEATS",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            "SEATS",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles: ",
              "TITLES",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Titles: ",
            "TITLES",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              "ANCESTRALWEAPONS",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            "ANCESTRALWEAPONS",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              "COATOFARMS",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            "COATOFARMS",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
